src/callbacks/apply.js

```diff
@@ -46,17 +67,29 @@ export default (credentialsCallback) => {
   return (to, from, source) => {
     const mode = source.mode();
 
-    const runNextFilter = () => Promise.resolve()
-      .then(() => {
-        if (mode === 1) {
-          return clean(to, from, source);
-        }
-        return smudge(to, from, source);
-      })
-      .then(
-        () => Error.CODE.OK,
-        () => Error.CODE.PASSTHROUGH
-      );
+    const runNextFilter = () => {
+      const event = mode === 1 ? 'clean' : 'smudge';
+      const startTime = Date.now();
+      return Promise.resolve()
+        .then(() => {
+          if (mode === 1) {
+            return clean(to, from, source);
+          }
+          return smudge(to, from, source);
+        })
+        .then(
+          () => {
+            const endTime = Date.now();
+            const deltaTime = endTime - startTime;
+            LFS_DEBUG.recordEvent(event, {
+              source: source.path(),
+              duration: deltaTime
+            });
+          },
+          () => Error.CODE.OK,
+          () => Error.CODE.PASSTHROUGH
+        );
+    };
```

Comment:
This doesn't look right. `.then` only takes 2 callbacks not 3 callbacks. If `finally` is available (check https://node.green/), do:
```javascript
.then(
  () => Error.Code.OK,
  () => Error.CODE.PASSTHROUGH
)
.finally(() => {/* debug goes here */});
```

Reply:
Checked `node.green`. Looks like the oldest version it is compatible with is `10.3.0`. Checked `nodegit` and we set `engines.node` to `">= 12.19.0 < 13 || >= 14.10.0"` so we should be good to go. Thanks for pointing this out. I had never seen this syntax before and thought this was `.then` taking a variable amount of functions to run but I should have looked up to confirm the syntax! Thanks again!